### PR TITLE
Flatpak manifest for com.endlessm.photos

### PR DESCRIPTION
**Pull request by cosimoc**
The Endless photo editor is an easy to use photo editing application with filters and customizable effects.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.endlessm.photos

File: com.endlessm.photos.json [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    ],
+    "modules": [
+        {
+            "name": "gfortran",
+            "buildsystem": "simple",
+            "build-commands": [ "/usr/lib/sdk/gfortran-62/install.sh" ]
+        },
+        {
+            "name": "lapack",
+            "buildsystem": "cmake",
```
[the remark this answers is not in the document]
Comment (cosimoc) [reply com.endlessm.photos.json]: Unfortunately it doesn't

**Comment by cosimoc** [comment]
@nedrichards Added those upstream and updated the manifest here.

**Comment by nedrichards** [comment]
bot, build com.endlessm.photos

**Comment by nedrichards** [comment]
Er, is the gfortran sdk not available on arm or aarch64? That's surprising. The only other app on flathub that seems to use it doesn't work on those architectures and https://github.com/flathub/com.github.wwmm.pulseeffects which uses that stuff seems to build it locally.

File: com.endlessm.photos.json [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        "--share=ipc", "--share=network",
+        "--socket=x11", "--socket=wayland",
+        "--device=dri",
+        "--filesystem=home",
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+     ],
+    "build-options": {
+        "env": {
+            "PATH": "/app/bin:/usr/bin:/usr/lib/sdk/gfortran-62/bin"
```
Comment (TingPing) [inline com.endlessm.photos.json]:
```json
"build-options": {
  "append-path": "/usr/lib/sdk/gfortran-62/bin"
}
```

File: com.endlessm.photos.json [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            "name": "gfortran",
+            "buildsystem": "simple",
+            "build-commands": [ "/usr/lib/sdk/gfortran-62/install.sh" ]
+        },
+        {
+            "name": "lapack",
+            "buildsystem": "cmake",
+            "builddir": true,
+            "config-opts": [
+                "-DCMAKE_INSTALL_PREFIX=/app",
```
Comment (TingPing) [inline com.endlessm.photos.json]: prefix and libdir are already set.

**Comment by cosimoc** [comment]
@TingPing Thanks, comments addressed.

File: com.endlessm.photos.json [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    "finish-args": [
+        "--share=ipc", "--share=network",
+        "--socket=x11", "--socket=wayland",
+        "--device=dri",
+        "--filesystem=home",
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+     ],
+    "build-options": {
+	"append-path": "/usr/bin:/usr/lib/sdk/gfortran-62/bin"
```
Comment (TingPing) [inline com.endlessm.photos.json]: You left `/usr/bin`.

**Comment by cosimoc** [comment]
Whoops, updated.

**Comment by nedrichards** [comment]
bot, build com.endlessm.photos

**Comment by nedrichards** [comment]
@cosimoc you either need to push a flathub.json to bar arm and aarch64 or wait until we work out exactly why there isn't a gfortran sdk for those architectures.

https://flathub.org/builds/#/builders/2/builds/1089

other than that, lgtm

**Comment by cosimoc** [comment]
@nedrichards With my Endless hat on, I'd rather wait, since we'd need an ARM build as well. But up to you, I can add the flathub.json if you think there's value in having it Intel-only for now.

**Comment by nedrichards** [comment]
With my Endless and Flathub hats, I'd rather wait as well.

**Comment by nedrichards** [comment]
OK, I've added a repo for gfortran since it wasn't actually being built at all in the new system and fixed the build for aarch64 and arm. It's building now, once that's done I'll try another test build of this.

**Comment by nedrichards** [comment]
bot, build com.endlessm.photos

**Comment by nedrichards** [comment]
I think something's still broken about gfortran on those architectures, needs more digging, but at least we have a system where we can perform builds.

**Comment by nedrichards** [comment]
bot, build com.endlessm.photos

File: python3-Pillow.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "python3-Pillow",
```
Comment (nedrichards) [inline python3-Pillow.json]: building this failed on arm. 
https://flathub.org/builds/#/builders/44/builds/172
I know that this works in other apps on flathub using a slightly different recipe: https://github.com/flathub/org.gabmus.hydrapaper/blob/68b9d617d937d3eb4137150e5e98546b41c7fb57/org.gabmus.hydrapaper.json#L31

**Comment by AsciiWolf** [comment]
bot, build com.endlessm.photos

**Comment by nedrichards** [comment]
bot, build com.endlessm.photos

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.endlessm.photos

**Pull request closed without merging**